### ISSUE TITLE
Docstring for AbstractDataFrame still reflects old syntax

### ISSUE THREAD

**Issue by kescobo** [JuliaData/DataFrames.jl#1919]
[See here](https://github.com/JuliaData/DataFrames.jl/blob/445ecd90caa81f0c3e647283734985b50e2f6ae8/src/abstractdataframe/abstractdataframe.jl#L41-L53).

I'd do a PR, but I don't feel I understand the new syntax and how it connects to the abstract type well enough to write anything coherent.

**Comment by bkamins** [comment]
Thank you for reporting. I will fix it.

**Comment by bkamins** [comment]
Let me here give a simplified view of the new syntax (not as precise as in https://juliadata.github.io/DataFrames.jl/stable/lib/indexing/, but giving an intuition):
* you always have to use two indices for `AbstractDataFrame` indexing
* passing a single column and a single row returns you a value
* passing a single column returns you a vector if you select multiple rows
* passing multiple columns and a single row returns you `DataFrameRow`
* passing multiple columns and multiple rows returns you an `AbstractDataFrame`
* `df.col` is a shorthand for `df[!, :col]` syntax
* allowed column selectors are: integer, vector of integer, vector of bool, `Symbol`, vector of `Symbol`, regex, `Not`, `All`, `Between`, `Colon`
* allowed row selectors are: integer, vector of integer, vector of bool, `Not`, `Colon`, `!`
* all multiple row selectors except `!` materialize a new object (`Vector` or `DataFrame`
* `!` row selector is non-copying (returns what is stored in an object, i.e. `Vector` and `DataFrame` for `DataFrame` and `SubArray` or `SubDataFrame` for `SubDataFrame`)
* broadcasting, broadcasting assignment and `setindex!` inherit the same rules; the crucial thing to remember is that `df[!, col]` and `df[!, cols]` replaces columns without copying them

In short `!` is a non-copying (and thus unsafe but faster) alternative to `:`.

**Comment by kescobo** [comment]
Makes sense to me! I think I actually did understand this, though I can't keep all the different access methods in my head. Nice to see it laid out here (and I still need to try out the `Not`, `All` etc syntax)

**Comment by bkamins** [comment]
`All` and `Between` is on master but not released yet.
